results/cmd/api/main.go

```diff
@@ -81,3 +81,27 @@ func (s *server) CreateTaskRun(ctx context.Context, req *pb.CreateTaskRunRequest
 	}
 	return taskrunFromClient, nil
 }
+
+// UpdateTaskRun receives TaskRun and FieldMask from client and uses them to update records in local Sqlite Server.
+func (s *server) UpdateTaskRun(ctx context.Context, req *pb.UpdateTaskRunRequest) (*pb.TaskRun, error) {
+	database := s.db
+
+	// Update the entire row in database based on uid of taskrun.
+	statement, err := database.Prepare("UPDATE taskrun SET name = ?, namespace = ?, taskrunlog = ? WHERE uid = ?")
+	if err != nil {
+		log.Printf("failed to update a existing taskrun: %v\n", err)
+		return nil, fmt.Errorf("failed to update a exsiting taskrun: %v", err)
```

Comment:
```suggestion
		return nil, fmt.Errorf("failed to update a exsiting taskrun: %w", err)
```

`%w` is generally preferred over `%v` when returning errors since it preserves underlying error information. See https://blog.golang.org/go1.13-errors for more details.

Reply:
Thanks for the advice

Reply:
done